src/StableMemory.mo

```diff
@@ -0,0 +1,93 @@
+/// Byte-level access to (virtual) _stable memory_.
+///
+/// This is a lightweight abstraction over IC _stable memory_ and supports persisting
+/// raw binary data across Motoko upgrades.
+/// Use of this module is fully compatible with Motoko's use of
+/// _stable variables_, whose persistence mechanism also uses (real) IC stable memory internally, but does not interfere with this API.
+///
+/// Each `load` operation loads from byte address `offset` in little-endian
+/// format using the natural bit-width of the type in question.
+/// The operation traps if attempting to read beyond the current stable memory size.
+///
+/// Each `store` operation stores to byte address `offset` in little-endian format using the natural bit-width of the type in question.
+/// The operation traps if attempting to write beyond the current stable memory size.
+///
+/// Text values can be handled by using `Text.decodeUtf8` and `Text.encodeUtf8`, in conjunction with `loadBlob` and `storeBlob`.
+///
+/// NB: The IC's actual stable memory size (`ic0.stable_size`) may exceed the
+/// page size reported by Motoko function `size()`.
+/// This is to  accommodate Motoko's stable variables.
+
+import Prim "mo:⛔";
+
+module {
+
+  /// Current size of the stable memory, in pages.
+  /// Each page is 16KiB (16384 bytes).
```

Comment:
but Wasm pages are 64k, right?

Reply:
Yeah, this is a mistake. I wonder where I got that from, and better check the implementation is correct...

Reply:
Good catch!

Reply:
```suggestion
  /// Each page is 64KiB (65536 bytes).
```